pkg/network/node/iptables.go

```diff
@@ -214,17 +223,29 @@ func (n *NodeIPTables) getNodeIPTablesChains() []Chain {
 }
 
 func (n *NodeIPTables) AddEgressIPRules(egressIP, mark string) error {
+	n.mu.Lock()
```

Comment:
syncIPTableRules() (holds n.mu lock) --> AddEgressIPRules()(tries to acquire same n.mu lock)
This should be okay if n.mu lock is re-entrant.
Is sync.Mutex lock re-entrant? Sample code https://play.golang.org/p/bCtXkdyuV5 shows locks in go are not re-entrant.

Reply:
@pravisankar fixed